Source/astcenccli_toplevel.cpp

```diff
@@ -581,102 +531,146 @@ int main(
 		}
 		else if (!strcmp(argv[5], "-medium"))
 		{
-			plimit_set = 25;
-			oplimit_set = 1.2f;
-			mincorrel_set = 0.75f;
-			dblimit_set = MAX(95.0f - 35.0f * log10_texels, 70.0f - 19.0f * log10_texels);
-			bmc_set = 75.0f;
-			maxiters_set = 2;
+			preset = ASTCENC_PRE_MEDIUM;
 		}
 		else if (!strcmp(argv[5], "-thorough"))
 		{
-			plimit_set = 100;
-			oplimit_set = 2.5f;
-			mincorrel_set = 0.95f;
-			dblimit_set = MAX(105.0f - 35.0f * log10_texels, 77.0f - 19.0f * log10_texels);
-			bmc_set = 95.0f;
-			maxiters_set = 4;
+			preset = ASTCENC_PRE_THOROUGH;
 		}
 		else if (!strcmp(argv[5], "-exhaustive"))
 		{
-			plimit_set = 1024;
-			oplimit_set = 1000.0f;
-			mincorrel_set = 0.99f;
-			dblimit_set = 999.0f;
-			bmc_set = 100.0f;
-			maxiters_set = 4;
+			preset = ASTCENC_PRE_EXHAUSTIVE;
 		}
 		else
 		{
 			printf("ERROR: Unknown search preset\n");
 			return 1;
 		}
 
-		if (decode_mode == ASTCENC_PRF_HDR)
-		{
-			ewp.rgb_power = 0.75f;
-			ewp.rgb_base_weight = 0.0f;
-			ewp.rgb_mean_weight = 1.0f;
+		argidx = 6;
+	}
+	else
+	{
+		// For decode the block size and preset are not needed.
+		argidx = 4;
+	}
 
-			ewp.alpha_power = 0.75f;
-			ewp.alpha_base_weight = 0.0f;
-			ewp.alpha_mean_weight = 1.0f;
+	unsigned int flags = 0;
 
-			dblimit_set = 999.0f;
+	while (argidx < argc)
+	{
+		if (!strcmp(argv[argidx], "-a"))
+		{
+			argidx += 2;
+			flags |= ASTCENC_FLG_USE_ALPHA_WEIGHT;
 		}
-		else if (decode_mode == ASTCENC_PRF_HDR_RGB_LDR_A)
+		// presets begin here
+		else if (!strcmp(argv[argidx], "-normal_psnr"))
 		{
-			ewp.rgb_power = 0.75f;
-			ewp.rgb_base_weight = 0.0f;
-			ewp.rgb_mean_weight = 1.0f;
-
-			ewp.alpha_base_weight = 0.05f;
-
-			dblimit_set = 999.0f;
+			argidx++;
+			flags |= ASTCENC_FLG_MAP_NORMAL;
 		}
+		else if (!strcmp(argv[argidx], "-normal_percep"))
+		{
+			argidx++;
 
-		argidx = 6;
+			flags |= ASTCENC_FLG_MAP_NORMAL;
+			flags |= ASTCENC_FLG_USE_PERCEPTUAL;
+		}
+		else if (!strcmp(argv[argidx], "-mask"))
+		{
+			argidx++;
+			flags |= ASTCENC_FLG_MAP_MASK;
+		}
+		else if (!strcmp(argv[argidx], "-linsrgb"))
+		{
+			argidx++;
+			flags |= ASTCENC_FLG_USE_LINEARIZED_SRGB;
+		}
+		else
+		{
+			argidx ++;
+		}
 	}
-	else
+
+	if (astcenc_init_config(profile, block_x, block_y, block_z, preset, flags, codec_configuration) != ASTCENC_SUCCESS)
 	{
-		// For decode the block size and preset are not needed.
-		argidx = 4;
+		printf ("ERROR: Failed to initialize configuration\n");
+		return 1;
 	}
 
+	return 0;
+}
+
+/**
+ * @brief Edit the astcenc_config
+ *
+ * @param argc
+ * @param argv
+ * @param op_mode
+ * @param array_size
+ * @param silentmode
+ * @param y_flip              Should this image be Y flipped?
+ * @param linearize_srgb      Should this image be converted to linear from sRGB?
+ * @param thread_count
+ * @param low_fstop
+ * @param high_fstop
+ * @param configuration
+ *
+ * @return 0 if everything is Okay, 1 if there is some error
+ */
+int edit_astcenc_config(
+		int argc,
```

Comment:
One though to make this parameter list less length would be to define a new cli_config struct for all of the additional options which are not part of astcenc_config.

Reply:
done